test/socket.jl

```diff
@@ -174,3 +174,56 @@ begin
     send(b, ip"::1", port, "Hello World")
     wait(tsk)
 end
+
+begin
+    default_port = UInt16(11011)
+    default_addr = IPv4("127.0.1.1")
+
+    sock = Base.TCPServer()
+    bind(sock,Base.InetAddr(default_addr,default_port))
+    listen(sock)
+
+    new_addr, new_port = getsockname(sock)
+
+    @test default_addr == new_addr
+    @test default_port == new_port
+    close(sock)
+end
+
+@unix_only begin
+    default_port = UInt16(21011)
+    default_addr = IPv6("::1")
+
+    sock = Base.TCPServer()
+    addr = Base.InetAddr(default_addr,default_port)
+    bind(sock,addr)
+    listen(sock)
+
+    new_addr, new_port = getsockname(sock)
+
+    @test default_addr == new_addr
+    @test default_port == new_port
+    close(sock)
+end
+
+begin
+    default_port = UInt16(11011)
+    default_addr = IPv4("127.0.1.1")
+
+    sock = Base.TCPServer()
+    bind(sock,Base.InetAddr(default_addr,default_port))
+    listen(sock)
+
+    @async begin
+        sleep(1)
+        ssock = connect(default_addr, default_port)
+    end
+
+    csock = accept(sock)
+    new_addr, new_port = getsockname(csock)
+
+    @test new_addr == IPv4("127.0.0.1")
```

Comment:
travis notes that you bound this to "127.0.1.1" on the other end, so that's the reported address for the tests